test/test_dataset/test_dataset.py

```diff
@@ -1,14 +1,14 @@
-import sys
+# -*- coding: utf-8 -*-
 import os
```

Comment:
The amount of tests for your branch that executes for this file is less than in master.

your branch:

```
$ .venv/bin/python3 -m pytest test/test_dataset/test_dataset.py  --log-level DEBUG -rA --tb=native -v
============================================================================ test session starts ============================================================================
platform linux -- Python 3.9.12, pytest-7.1.1, pluggy-1.0.0 -- /home/iwana/sw/d/github.com/iafork/rdflib.cleanish/.venv/bin/python3
cachedir: .pytest_cache
rootdir: /home/iwana/sw/d/github.com/iafork/rdflib.cleanish, configfile: tox.ini
plugins: subtests-0.7.0, cov-3.0.0
collected 4 items                                                                                                                                                           

test/test_dataset/test_dataset.py::test_graph_aware[BerkeleyDB] PASSED                                                                                                [ 25%]
test/test_dataset/test_dataset.py::test_default_graph[BerkeleyDB] PASSED                                                                                              [ 50%]
test/test_dataset/test_dataset.py::test_not_union[BerkeleyDB] PASSED                                                                                                  [ 75%]
test/test_dataset/test_dataset.py::test_iter[BerkeleyDB] PASSED                                                                                                       [100%]

============================================================================= warnings summary ==============================================================================
.venv/lib64/python3.9/site-packages/_pytest/fixtures.py:227
  /home/iwana/sw/d/github.com/iafork/rdflib.cleanish/.venv/lib64/python3.9/site-packages/_pytest/fixtures.py:227: UserWarning: Code: _pytestfixturefunction is not defined in namespace XSD
    fixturemarker: Optional[FixtureFunctionMarker] = getattr(

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
================================================================================== PASSES ===================================================================================
========================================================================== short test summary info ==========================================================================
PASSED test/test_dataset/test_dataset.py::test_graph_aware[BerkeleyDB]
PASSED test/test_dataset/test_dataset.py::test_default_graph[BerkeleyDB]
PASSED test/test_dataset/test_dataset.py::test_not_union[BerkeleyDB]
PASSED test/test_dataset/test_dataset.py::test_iter[BerkeleyDB]
======================================================================= 4 passed, 1 warning in 4.12s ========================================================================
```

master branch:

```
$ .venv/bin/python3 -m pytest test/test_dataset/test_dataset.py  --log-level DEBUG -rA --tb=native -v
============================================================================ test session starts ============================================================================
platform linux -- Python 3.9.12, pytest-7.1.1, pluggy-1.0.0 -- /home/iwana/d.x/github.com/RDFLib/rdflib/.venv/bin/python3
cachedir: .pytest_cache
rootdir: /home/iwana/d.x/github.com/RDFLib/rdflib, configfile: tox.ini
plugins: subtests-0.7.0, cov-3.0.0
collected 12 items                                                                                                                                                          

test/test_dataset/test_dataset.py::DatasetTestCase::testDefaultGraph PASSED                                                                                           [  8%]
test/test_dataset/test_dataset.py::DatasetTestCase::testGraphAware PASSED                                                                                             [ 16%]
test/test_dataset/test_dataset.py::DatasetTestCase::testIter PASSED                                                                                                   [ 25%]
test/test_dataset/test_dataset.py::DatasetTestCase::testNotUnion PASSED                                                                                               [ 33%]
test/test_dataset/test_dataset.py::t0::testDefaultGraph PASSED                                                                                                        [ 41%]
test/test_dataset/test_dataset.py::t0::testGraphAware PASSED                                                                                                          [ 50%]
test/test_dataset/test_dataset.py::t0::testIter PASSED                                                                                                                [ 58%]
test/test_dataset/test_dataset.py::t0::testNotUnion PASSED                                                                                                            [ 66%]
test/test_dataset/test_dataset.py::t1::testDefaultGraph SKIPPED (No SPARQL endpoint for SPARQLUpdateStore (tests skipped))                                            [ 75%]
test/test_dataset/test_dataset.py::t1::testGraphAware SKIPPED (No SPARQL endpoint for SPARQLUpdateStore (tests skipped))                                              [ 83%]
test/test_dataset/test_dataset.py::t1::testIter SKIPPED (No SPARQL endpoint for SPARQLUpdateStore (tests skipped))                                                    [ 91%]
test/test_dataset/test_dataset.py::t1::testNotUnion SKIPPED (No SPARQL endpoint for SPARQLUpdateStore (tests skipped))                                                [100%]

================================================================================== PASSES ===================================================================================
========================================================================== short test summary info ==========================================================================
PASSED test/test_dataset/test_dataset.py::DatasetTestCase::testDefaultGraph
PASSED test/test_dataset/test_dataset.py::DatasetTestCase::testGraphAware
PASSED test/test_dataset/test_dataset.py::DatasetTestCase::testIter
PASSED test/test_dataset/test_dataset.py::DatasetTestCase::testNotUnion
PASSED test/test_dataset/test_dataset.py::t0::testDefaultGraph
PASSED test/test_dataset/test_dataset.py::t0::testGraphAware
PASSED test/test_dataset/test_dataset.py::t0::testIter
PASSED test/test_dataset/test_dataset.py::t0::testNotUnion
SKIPPED [1] test/test_dataset/test_dataset.py:134: No SPARQL endpoint for SPARQLUpdateStore (tests skipped)
SKIPPED [1] test/test_dataset/test_dataset.py:81: No SPARQL endpoint for SPARQLUpdateStore (tests skipped)
SKIPPED [1] test/test_dataset/test_dataset.py:173: No SPARQL endpoint for SPARQLUpdateStore (tests skipped)
SKIPPED [1] test/test_dataset/test_dataset.py:160: No SPARQL endpoint for SPARQLUpdateStore (tests skipped)
======================================================================= 8 passed, 4 skipped in 4.18s ========================================================================
```

Reply:
That's .. bizarre. I think the current master output is misleading, it seems that `default` _is_ being tested even though it's filtered out of the assembled list of pluginstores, by virtue of the `store = "default"` at https://github.com/RDFLib/rdflib/blob/6b356a4eacde1280f78b4ed966e8a499a727189a/test/test_dataset/test_dataset.py#L30

If neither BerkeleyDB nor SPARQLUpdateStore are available, the output (locally) is:
```
test/test_dataset/test_dataset.py::t0::testDefaultGraph SKIPPED (Dependencies for store 'BerkeleyDB' not available!)                                                                                                           [  8%]
test/test_dataset/test_dataset.py::t0::testGraphAware SKIPPED (Dependencies for store 'BerkeleyDB' not available!)                                                                                                             [ 16%]
test/test_dataset/test_dataset.py::t0::testIter SKIPPED (Dependencies for store 'BerkeleyDB' not available!)                                                                                                                   [ 25%]
test/test_dataset/test_dataset.py::t0::testNotUnion SKIPPED (Dependencies for store 'BerkeleyDB' not available!)                                                                                                               [ 33%]
test/test_dataset/test_dataset.py::t1::testDefaultGraph SKIPPED (No SPARQL endpoint for SPARQLUpdateStore (tests skipped))                                                                                                     [ 41%]
test/test_dataset/test_dataset.py::t1::testGraphAware SKIPPED (No SPARQL endpoint for SPARQLUpdateStore (tests skipped))                                                                                                       [ 50%]
test/test_dataset/test_dataset.py::t1::testIter SKIPPED (No SPARQL endpoint for SPARQLUpdateStore (tests skipped))                                                                                                             [ 58%]
test/test_dataset/test_dataset.py::t1::testNotUnion SKIPPED (No SPARQL endpoint for SPARQLUpdateStore (tests skipped))                                                                                                         [ 66%]
test/test_dataset/test_dataset.py::DatasetTestCase::testDefaultGraph PASSED                                                                                                                                                    [ 75%]
test/test_dataset/test_dataset.py::DatasetTestCase::testGraphAware PASSED                                                                                                                                                      [ 83%]
test/test_dataset/test_dataset.py::DatasetTestCase::testIter PASSED                                                                                                                                                            [ 91%]
test/test_dataset/test_dataset.py::DatasetTestCase::testNotUnion PASSED                                                                                                                                                        [100%]
``` 

So, to make these match (and I'm disinclined to take the legacy comment “# these are tested by default” at face value), I should just remove "default" from the list of to-be-excluded pluginstores:

```
test/test_dataset/test_dataset.py::test_graph_aware[default] PASSED                                                                                                                                                            [  8%]
test/test_dataset/test_dataset.py::test_graph_aware[BerkeleyDB] PASSED                                                                                                                                                         [ 16%]
test/test_dataset/test_dataset.py::test_graph_aware[SPARQLUpdateStore] PASSED                                                                                                                                                  [ 25%]
test/test_dataset/test_dataset.py::test_default_graph[default] PASSED                                                                                                                                                          [ 33%]
test/test_dataset/test_dataset.py::test_default_graph[BerkeleyDB] PASSED                                                                                                                                                       [ 41%]
test/test_dataset/test_dataset.py::test_default_graph[SPARQLUpdateStore] PASSED                                                                                                                                                [ 50%]
test/test_dataset/test_dataset.py::test_not_union[default] PASSED                                                                                                                                                              [ 58%]
test/test_dataset/test_dataset.py::test_not_union[BerkeleyDB] PASSED                                                                                                                                                           [ 66%]
test/test_dataset/test_dataset.py::test_not_union[SPARQLUpdateStore] PASSED                                                                                                                                                    [ 75%]
test/test_dataset/test_dataset.py::test_iter[default] PASSED                                                                                                                                                                   [ 83%]
test/test_dataset/test_dataset.py::test_iter[BerkeleyDB] PASSED                                                                                                                                                                [ 91%]
test/test_dataset/test_dataset.py::test_iter[SPARQLUpdateStore] PASSED                                                                                                                                                         [100%]
```

Reply:
Okay I checked the coverage summary for this now and it looks more or less the same as in master.